src/Core/SettingsEnums.cpp

```diff
@@ -99,6 +100,12 @@ IMPLEMENT_SETTING_MULTI_ENUM(MySQLDataTypesSupport, ErrorCodes::UNKNOWN_MYSQL_DA
      {"date2Date32", MySQLDataTypesSupport::DATE2DATE32},
      {"date2String", MySQLDataTypesSupport::DATE2STRING}})
 
+IMPLEMENT_SETTING_MULTI_ENUM(PostgreSQLDataTypesSupport, ErrorCodes::UNKNOWN_POSTGRESQL_DATATYPES_SUPPORT_LEVEL,
+    {{"decimal",    PostgreSQLDataTypesSupport::DECIMAL},
+     {"datetime64", PostgreSQLDataTypesSupport::DATETIME64},
+     {"date2Date32", PostgreSQLDataTypesSupport::DATE2DATE32},
+     {"date2String", PostgreSQLDataTypesSupport::DATE2STRING}})
```

Comment:
hm, but these data type conversion options are not implemented for postgresql as far as I see, then no need to add them in my opinion

Reply:
Turns out it's needed for `DECLARE_SETTINGS_TRAITS(PostgreSQLSettingsTraits, LIST_OF_POSTGRESQL_SETTINGS)` in `PosgreSQL/PostgreSQLSettings.h`:

```
/home/ubuntu/kkz-ClickHouse/src/Storages/PostgreSQL/PostgreSQLSettings.h:27:1: error: unknown type name 'SettingFieldPostgreSQLDataTypesSupport'; did you mean 'SettingFieldMySQLDataTypesSupport'?
DECLARE_SETTINGS_TRAITS(PostgreSQLSettingsTraits, LIST_OF_POSTGRESQL_SETTINGS)
^
/home/ubuntu/kkz-ClickHouse/src/Core/BaseSettings.h:898:5: note: expanded from macro 'DECLARE_SETTINGS_TRAITS'
    DECLARE_SETTINGS_TRAITS_COMMON(SETTINGS_TRAITS_NAME, LIST_OF_SETTINGS_MACRO, 0)
    ^
/home/ubuntu/kkz-ClickHouse/src/Core/BaseSettings.h:910:36: note: expanded from macro 'DECLARE_SETTINGS_TRAITS_COMMON'
            LIST_OF_SETTINGS_MACRO(DECLARE_SETTINGS_TRAITS_, SKIP_ALIAS) \
                                   ^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsEnums.h:144:1: note: 'SettingFieldMySQLDataTypesSupport' declared
here
DECLARE_SETTING_MULTI_ENUM(MySQLDataTypesSupport)
^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsFields.h:544:5: note: expanded from macro 'DECLARE_SETTING_MULTI_ENUM'
    DECLARE_SETTING_MULTI_ENUM_WITH_RENAME(ENUM_TYPE, ENUM_TYPE)
    ^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsFields.h:557:11: note: expanded from macro 'DECLARE_SETTING_MULTI_ENUM_WITH_RENAME'
    using SettingField##NEW_NAME = SettingFieldMultiEnum<ENUM_TYPE, SettingField##NEW_NAME##Traits>;
          ^
<scratch space>:31:1: note: expanded from here
SettingFieldMySQLDataTypesSupport
```

Reply:
It turns out that we do need these:

```
In file included from /home/ubuntu/kkz-ClickHouse/src/Storages/PostgreSQL/StorageMaterializedPostgreSQL.cpp:34:
In file included from /home/ubuntu/kkz-ClickHouse/src/Storages/StoragePostgreSQL.h:7:
/home/ubuntu/kkz-ClickHouse/src/Storages/PostgreSQL/PostgreSQLSettings.h:27:1: error: unknown type name 'Settin
gFieldPostgreSQLDataTypesSupport'; did you mean 'SettingFieldMySQLDataTypesSupport'?
DECLARE_SETTINGS_TRAITS(PostgreSQLSettingsTraits, LIST_OF_POSTGRESQL_SETTINGS)
^
/home/ubuntu/kkz-ClickHouse/src/Core/BaseSettings.h:898:5: note: expanded from macro 'DECLARE_SETTINGS_TRAITS'
    DECLARE_SETTINGS_TRAITS_COMMON(SETTINGS_TRAITS_NAME, LIST_OF_SETTINGS_MACRO, 0)
    ^
/home/ubuntu/kkz-ClickHouse/src/Core/BaseSettings.h:910:36: note: expanded from macro 'DECLARE_SETTINGS_TRAITS_
COMMON'
            LIST_OF_SETTINGS_MACRO(DECLARE_SETTINGS_TRAITS_, SKIP_ALIAS) \
                                   ^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsEnums.h:144:1: note: 'SettingFieldMySQLDataTypesSupport' declared
here
DECLARE_SETTING_MULTI_ENUM(MySQLDataTypesSupport)
^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsFields.h:544:5: note: expanded from macro 'DECLARE_SETTING_MULTI_E
NUM'
    DECLARE_SETTING_MULTI_ENUM_WITH_RENAME(ENUM_TYPE, ENUM_TYPE)
    ^
/home/ubuntu/kkz-ClickHouse/src/Core/SettingsFields.h:557:11: note: expanded from macro 'DECLARE_SETTING_MULTI_
ENUM_WITH_RENAME'
    using SettingField##NEW_NAME = SettingFieldMultiEnum<ENUM_TYPE, SettingField##NEW_NAME##Traits>;
          ^
<scratch space>:20:1: note: expanded from here
```

Reply:
looks like you just need to delete this line `M(PostgreSQLDataTypesSupport, postgresql_datatypes_support_level, 0, ..`  which adds unsupported setting